### ISSUE TITLE
Calling OS.shell_open() from inside a deferred_call() function messes up the call queue

### ISSUE THREAD

**Issue by salalex** [godotengine/godot#4353]
**Operating system or device:**
Windows 7 Pro SP1
Intel Core i7 2.4GHz, 16GB RAM
Godot 2.0.2.stable.official

**Issue description** (what happened, and what was expected):
If you call `OS.shell_open()` from a function which was called by `deferred_call()` inside a `button_pressed()` signal handler, the execution of the deferred function itself is restarted al least one time, and often several times, especially if you keep moving the mouse after clicking the button.
I would expect the deferred function to be called only once for every button press.

**Steps to reproduce:**
1. Create a function named `deferred_open()`
2. Inside `deferred_open()` add `print("Deferred open")` followed by `OS.shell_open("anything")`
3. Add a button
4. Connect the `button_pressed()` signal to a function, let's say `_on_Button_pressed()`
5. Inside `_on_Button_pressed()`  add `call_deferred("deferred_open")`
6. Run the scene and click on the button: see that "Deferred open" is printed two times (or more)

**Link to minimal example project** (optional but very welcome):
I tried to upload a zipfile with an example project, but I keep getting the error "_Unfortunately, we don’t support that file type. Try again with a PNG, GIF, JPG, DOCX, PPTX, XLSX, TXT, PDF, or ZIP._" (!?!), so here is an external link: http://www.olomatrix.org/godot/ShellOpenBug.zip

**Comment by bojidar-bg** [comment]
@salalex ~ Sorry for bothering you, but... does the same happen if you comment out the shell_open line? :smile:

**Comment by salalex** [comment]
@bojidar-bg No, in that case it does not happen to me :blush: It seems that it's really `shell_open()` that triggers the problem. 
Besides, upon further testing, and contrary to what I wrote in the example project, I found that it's not necessary to have `call_deferred()` called from the `button_pressed() ` handler: it happens even if you call `call_deferred()` from `_ready()` and move the mouse while the call is being made.

**Comment by reduz** [comment]
I can't reproduce this on Linux, is this Windows only?

**Comment by vnen** [comment]
I can reproduce on Windows.

**Comment by salalex** [comment]
I also tried on Linux (Ubuntu 14.02.2 LTS) and I can confirm it doesn't happen there: it seems it is really a Windows-only issue.

**Comment by salalex** [comment]
Any news on this issue?

**Comment by salalex** [comment]
Did anything change in 3.0 regarding shell_open()?

**Comment by vnen** [comment]
@salalex I used your test project (after an update for 3.0) and cannot reproduce this issue. It is still a problem in Godot 2.1 though.

**Comment by salalex** [comment]
Oh, well, another reason to upgrade to 3.0 as soon as possible!
Thank you

**Comment by aaronfranke** [comment]
It might be worth closing this issue, it seems like a non-critical bug and the 2.1 branch is not being actively developed anymore (mostly just updates to keep it working on modern smartphones).

If you need to have this working on Godot 2.1, you could also consider using Linux (it seems everyone here has reported it works fine on Linux).

**Comment by akien-mga** [comment]
Indeed, thanks for the review. Closing as fixed in 3.0.
